File: go-client-sdk/pkg/models/operations/getcouponscouponid.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetCouponsCouponIDPathParams struct {
	CouponID string `pathParam:"style=simple,explode=false,name=coupon_id"`
}

type GetCouponsCouponIDRequest struct {
	PathParams GetCouponsCouponIDPathParams
}

type GetCouponsCouponIDResponse struct {
	ContentType string
	Coupon      *shared.Coupon
	StatusCode  int
}

File: typescript-client-sdk/src/sdk/models/operations/getcouponscouponid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetCouponsCouponIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=coupon_id" })
  couponId: string;
}


export class GetCouponsCouponIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCouponsCouponIdPathParams;
}


export class GetCouponsCouponIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  coupon?: shared.Coupon;

  @SpeakeasyMetadata()
  statusCode: number;
}
